libs/lib-mongodb/src/types/types.ts

```diff
@@ -33,29 +42,36 @@ export function baseUri(options: BaseMongoConfig) {
  *
  * For use by both storage and mongo module.
  */
-export function normalizeMongoConfig(options: BaseMongoConfigDecoded) {
-  let uri = urijs.parse(options.uri);
+export function normalizeMongoConfig(options: BaseMongoConfigDecoded): NormalizedMongoConfig {
+  let uri: ConnectionURI;
 
-  const database = options.database ?? uri.path?.substring(1) ?? '';
+  try {
+    uri = new ConnectionURI(options.uri);
+  } catch (error) {
+    throw new ServiceError(
+      ErrorCode.PSYNC_S1109,
+      `MongoDB connection: invalid URI ${error instanceof Error ? `- ${error.message}` : ''}`
+    );
+  }
 
-  const userInfo = uri.userinfo?.split(':');
+  const database = options.database ?? uri.pathname.split('/')[1] ?? '';
+  const username = options.username ?? uri.username;
+  const password = options.password ?? uri.password;
 
-  const username = options.username ?? userInfo?.[0];
-  const password = options.password ?? userInfo?.[1];
+  uri.password = '';
+  uri.username = '';
 
   if (database == '') {
     throw new ServiceError(ErrorCode.PSYNC_S1105, `MongoDB connection: database required`);
   }
 
-  delete uri.userinfo;
-
   const lookupOptions: LookupOptions = {
     reject_ip_ranges: options.reject_ip_ranges ?? []
   };
-  const lookup = makeHostnameLookupFunction(uri.host ?? '', lookupOptions);
+  const lookup = makeHostnameLookupFunction(uri.hosts[0] ?? '', lookupOptions);
```

Comment:
This will need to be extended to validate all hosts instead of just the first one, but we can handle that after this PR is merged.